### ISSUE TITLE
(Linux) IVPN 3.8.20 breaks dig command when specifying DNS server

### ISSUE THREAD

**Issue by Kartikeya-IYER** [ivpn/desktop-app#167]
# Bug report

## Describe your environment

* Device: Odroid HC4 (an arm64 SoC device)  
* OS name and version: Ubuntu 22.04 (arm64 build provided by Armbian) 
* IVPN app version: 3.8.20  

## Describe the problem
After upgrading to IVPN 3.8.20, the following `dig` command no longer works. It used to work on the previous IVPN GA release v3.8.7.

The dig command is:
```
dig +short myip.opendns.com @resolver1.opendns.com
```

I use this command to check what my external IP is.

With 3.8.20 this command no longer works when IVPN is active. It does work when IVPN is disabled (via GUI). The same behavior persists whether or not I use a custom DNS with IVPN (if I use a custom DNS I usually use Quad9's DNS `9.9.9.9`)

On the previous IVPN version 3.8.7 this command worked when IVPN was enabled and active.

Note that when dig errors out when the VPN is active, I can still traceroute to `opendns.com` until it reaches the opendns DMZ (further traceroute packets seem to get blocked after it reaches the first `opendns.com` server, and this traceroute behavior is the same even with VPN disabled). 

I am wondering whether one of the DNS hardening changes has broken dig, and when I have some time to spare I will embark upon an exercise to see if I can narrow down which commit caused this to occur. Unfortunately, due to issue #21 it is a bit of an involved process to switch back and forth between IVPN versions/builds so this detective work might take a while.

### Steps to reproduce:

1. Run the dig command above without IVPN active. Observe that it works and reports the external IP address.
2. Install or build the previous IVPN release 3.8.7 on Linux (either Debian or Ubuntu, as these were what I tried).
3. Enable the VPN. My VPN settings used multihop (entry server in LU, exit server in CH) but I don't think this matters.
4. Run the above dig command with the VPN enabled.
5. Observe that the dig command works and that the external IP address it reports is the IP address through the VPN.
6. Repeat steps 2, 3, and 4 above for the latest IVPN release 3.8.20
7. Observe that the dig command no longer works, and instead reports an error when the VPN is active.

### Observed Results:

When the VPN is running, the dig command spins a while and then errors out with:
```
;; connection timed out; no servers could be reached
```

### Expected Results:

dig succeeds and reports VPN-derived IP address.

### Relevant Code:

* When the VPN is **not** active:
    ```
    root@odroidhc4:~# dig +short myip.opendns.com @resolver1.opendns.com
    <REDACTED>.<REDACTED>.176.71
    
    root@odroidhc4:~# traceroute opendns.com
    traceroute to opendns.com (146.112.62.105), 30 hops max, 60 byte packets
     1  BeastlyE3000-Repeater (192.168.2.1)  0.987 ms  1.488 ms  1.716 ms
     2  192.168.1.1 (192.168.1.1)  6.166 ms  8.045 ms  8.708 ms
     3  192.168.0.1 (192.168.0.1)  8.608 ms  9.066 ms  9.321 ms
     4  <REDACTED PROVIDER DETAILS>  24.325 ms  23.429 ms  24.989 ms
     5  <REDACTED PROVIDER DETAILS>  33.507 ms  37.948 ms  33.660 ms
     6  <REDACTED PROVIDER DETAILS>  56.246 ms  49.391 ms  49.599 ms
     7  gi0-1.rtr1.ash.opendns.com (206.126.236.128)  48.881 ms  50.887 ms  48.969 ms
     8  * * *
     9  * * *
    10  * * *
    11  * * *
    12  * * *
    13  *^C
    
    root@odroidhc4:~# 
    
    ```

* On IVPN 3.8.20 with VPN **active**:
    ```
    root@odroidhc4:~# dig +short myip.opendns.com @resolver1.opendns.com
    ;; connection timed out; no servers could be reached
    
    root@odroidhc4:~# traceroute opendns.com
    traceroute to opendns.com (146.112.62.105), 30 hops max, 60 byte packets
     1  172.16.0.1 (172.16.0.1)  151.306 ms  151.708 ms  153.954 ms
     2  179.43.183.245 (179.43.183.245)  157.180 ms  160.754 ms  161.997 ms
     3  185.227.149.8 (185.227.149.8)  161.875 ms  162.914 ms  165.023 ms
     4  185.227.149.17 (185.227.149.17)  175.465 ms  178.120 ms  179.482 ms
     5  100ge0-2.core3.fra1.he.net (184.104.204.249)  181.786 ms  183.530 ms  185.010 ms
     6  * port-channel2.core2.par2.he.net (184.104.198.57)  168.276 ms  180.440 ms
     7  100ge10-2.core1.ash1.he.net (184.105.213.173)  277.973 ms  281.775 ms  283.837 ms
     8  gi0-1.rtr1.ash.opendns.com (206.126.236.128)  289.150 ms  289.354 ms  289.622 ms
     9  * * *
    10  * * *
    11  * * *
    12  * * *
    13  * * *
    14  *^C
    
    root@odroidhc4:~# 
    
    ```

**Comment by Kartikeya-IYER** [comment]
**Workarounds**: I have been able to use one of the following to determine my external IP address when IVPN 3.8.20 is active:
```
curl -4 icanhazip.com
curl https://ipecho.net/plain;echo
curl https://ipinfo.io/ip;echo
```
Basically, anything that doesn't involve queries to DNS servers (`dig`, `host`, etc) should work, although my preference is to use actual DNS queries for this.

**Comment by stenya** [comment]
@Kartikeya-IYER I suppose, the dig command you use is trying to send DNS requests to some external server (not configured by IVPN).
If so - such behavior is expected. IVPN firewall allow DNS requests only to one server which is configured by IVPN Client.

**Comment by Kartikeya-IYER** [comment]
@stenya Yep, the dig command I posted uses `resolver1.opendns.com` as the DNS server to query. 

I guess that - as you said - this is expected behaviour to be able to avoid DNS leaks. As I mentioned in the bug report, I did suspect that dig might have broken due to the DNS hardening changes in 3.8.20 which I saw in the changelog.

I will experiment whether using dig with a custom DNS will work.

I am closing this ticket.
